File: Clojure/src/Histogram.clj
(ns Histogram)

(defn hist [s]
	(apply str
		(interpose "\n"
			(map
				(fn [[key value]]
					(str (format "%s  %-6d" key value) (apply str (repeat value "*"))))
				(filter
					(fn [[_ value]]
						(< 0 value))
					(map
						(fn [[key _]]
							[key (- (count s) (count (.replaceAll s key "")))])
						{"u" 0 "w" 0 "x" 0 "z" 0}))))))

File: Kotlin/src/main/kotlin/Histogram.kt
object Histogram
{
	fun hist(s: String): String
	{
		val count = "abcdefghijklmnopqrstuvwxyz".associate { letter -> Pair(letter, 0) }.toMutableMap()
		for (letter in s)
		{
			count[letter] = count[letter]!!.plus(1)
		}
		val res = mutableListOf<String>()
		for (letter in "uwxz")
		{
			if (count[letter]!! > 0)
			{
				val num = "%-6d".format(count[letter])
				val stars = "*".repeat(count[letter]!!)
				res.add("$letter  $num$stars")
			}
		}

		return res.joinToString("\n")
	}
}
